snuba/query/processors/tagsmap.py

```diff
@@ -103,6 +104,14 @@ def __is_optimizable(
             )
         return None
 
+    def __has_tags(self, expression: Expression) -> bool:
```

Comment:
I'm not sure if this is a good idea or not, but making this `expression: Optional[Expression])` would allow you to remove a bunch of the left side expressions from `if x and self.__has_tags(x)` below.

Reply:
No strong opinion since it is a private method. Will make the change